pandas/tests/io/test_html.py

```diff
@@ -1280,6 +1280,8 @@ def seek(self, offset):
             def seekable(self):
                 return True
 
+            # GH 48855 suppressing false positives
+            # pylint: disable-next=non-iterator-returned
```

Comment:
Can this be corrected such that the check and existing test passes?

Reply:
I'll give that a shot this evening.

Reply:
After reviewing #17975 ( issue prompting this test), I added `__next__` and changed `__iter__` to remove the warning from pylint while letting existing test pass.